frontend/javascripts/admin/dataset/dataset_upload_view.js

```diff
@@ -174,24 +175,34 @@ class DatasetUploadView extends React.Component<PropsWithFormAndRouter, State> {
     }
   }
 
+  componentWillUnmount() {
+    this.unblockHistory();
+  }
+
+  unblockHistory() {
+    window.onbeforeunload = null;
+    if (this.unblock != null) {
+      this.unblock();
+    }
+  }
+
   isDatasetManagerOrAdmin = () =>
     this.props.activeUser &&
     (this.props.activeUser.isAdmin || this.props.activeUser.isDatasetManager);
 
   handleSubmit = async formValues => {
     const { activeUser } = this.props;
-    const pathNameAtSubmit = window.location.pathname;
 
     if (activeUser != null) {
       Toast.info("Uploading dataset");
       this.setState({
         isUploading: true,
       });
-      const beforeUnload = action => {
+      const beforeUnload = (newLocation, action) => {
         // Only show the prompt if this is a proper beforeUnload event from the browser
         // or the pathname changed
         // This check has to be done because history.block triggers this function even if only the url hash changed
-        if (action === undefined || pathNameAtSubmit !== window.location.pathname) {
```

Comment:
@MichaelBuessemeyer Could you review this PR in general and pay special attention to how the "warn when leaving unsaved page" mechanism is implemented? Initially, I copied the code from "dataset_upload_view", but that didn't really work for me (both for the new page as well as the old page). Then, I found that it differs from the implementation for the wk tracing view (which seems to work) and decided that this should be the way to go (see https://github.com/scalableminds/webknossos/pull/5948/files#diff-3da8d2bc2de3cc52dea2773d430e07e10e55dd54993bab3c5f5669df75b5a7b1L142).
Do you remember the reason for using `pathNameAtSubmit`? I don't see why this would make sense.

For the record, this is what I tested:
- upload a dataset and during upload, navigate away --> I only got a warning with the newest changes from this PR. before that, wk happily navigated away.
- in the dataset edit view, I changed the form and then tried to navigate away via:
  - save
  - cancel
  - use wk navbar
  - use page refresh
  ---> when there are unsaved changes, wk should warn me. canceling the dialog should navigate away. clicking "ok", should abort the navigation. using the form's "cancel" button is the only exception (in that case no popup should appear).

Could you doublecheck this behavior, @MichaelBuessemeyer? Maybe in firefox, since I used Chrome myself. Thank you!

Reply:
for reference, the docs: https://github.com/remix-run/history/blob/v4/docs/Blocking.md

Reply:
> Do you remember the reason for using pathNameAtSubmit? I don't see why this would make sense.

I does make sense to me either 🤷‍♂️. Maybe because the interface of the form changed, I somehow needed to fix the old code and this is what I went for. But comparing with the `pathBeforeSubmit` is weird. The code reads like it doesn't block the user from navigating away, right?

I just managed to produce a bug (in Chrome MacOS):
1. Got to Upload datasets.
2. Upload the ROI dataset.
3. While uploading, change the url to `http://localhost:9000/`.
4. Then the browser prompts correctly about unsaved changes.
5. I click "Stop / Stay" and the page still changes to the dashboard and the upload stops :/

Reply:
Ok I just checked it again. Looks like I did not perform my first test properly.
There is an up to date version:
- The before unload seems to work on MacOS Chrome on the upload dataset page. It successfully continues the upload if the user cancels it.
- For Firefox on MacOS the Upload gets stuck after telling not to move away from the page. I hope it is easy to reproduce.